### PR TITLE
[15.0][ADD] mrp_subcontracting_lot_number_propagation

### PR DESCRIPTION
**Pull request by sebalix**
Depends on:
- https://github.com/OCA/manufacture/pull/843

Only the second commit should be reviewed.

Ref dec-120

### CODE REVIEW

File: mrp_lot_number_propagation/models/mrp_bom.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    @api.constrains("lot_number_propagation")
+    def _check_propagate_lot_number(self):
+        for bom in self:
+            if not bom.lot_number_propagation:
+                continue
+            if not bom.bom_line_ids.filtered("propagate_lot_number"):
+                raise ValidationError(
+                    _(
+                        "With 'Lot Number Propagation' enabled, a line has "
+                        "to be configured with the 'Propagate Lot Number' option."
```
Comment (i-vyshnevska) [inline mrp_lot_number_propagation/models/mrp_bom.py]:
```suggestion
                        "With 'Lot Number Propagation' enabled, a line has on order id {}"
                        "to be configured with the 'Propagate Lot Number' option.".format(bom.id)
```

**Review by ivantodorovich — APPROVED** [review]

File: mrp_subcontracting_lot_number_propagation/models/stock_move_line.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        string="Is lot editable?",
+        compute="_compute_is_lot_id_editable",
+        help="Technical field to define if the 'lot_id' field is editable.",
+    )
+
+    @api.depends("lot_id")
+    def _compute_is_lot_id_editable(self):
+        for line in self:
+            line.is_lot_id_editable = True
+            mo_ids = line.move_id.move_orig_ids.mapped("production_id")
```
Comment (sebalix) [inline mrp_subcontracting_lot_number_propagation/models/stock_move_line.py]: So we know it's not a list of IDs but a recordset
```suggestion
            mos = line.move_id.move_orig_ids.mapped("production_id")
```

**Comment by grindtildeath** [comment]
@OCA/manufacturing-maintainers Can we please keep this one open?

**Review by grindtildeath — APPROVED** [review]
Can you please set maturity level to Beta and squash?

**Pull request merged**